firebase-installations/src/main/java/com/google/firebase/installations/FirebaseInstallations.java

```diff
@@ -192,14 +188,15 @@ private static boolean persistedFidMissingOrInErrorState(PersistedFidEntry persi
   }
 
   @NonNull
-  private <F, T> Continuation<F, T> call(@NonNull Supplier<T> supplier) {
+  private <F, T> Continuation<F, T> awaitFidRegistration(@NonNull Supplier<T> supplier) {
```

Comment:
instead of doing this weird out of band latch thing, we could just copy the Tasks.await implementation that uses a real listener

Reply:
PTAL.